src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/PageApplicationModelProviderContext.cs

```diff
@@ -10,9 +11,25 @@ namespace Microsoft.AspNetCore.Mvc.ApplicationModels
     /// </summary>
     public class PageApplicationModelProviderContext
     {
+        public PageApplicationModelProviderContext(PageActionDescriptor descriptor, TypeInfo pageTypeInfo)
+        {
+            ActionDescriptor = descriptor;
+            PageType = pageTypeInfo;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="PageActionDescriptor"/>.
+        /// </summary>
+        public PageActionDescriptor ActionDescriptor { get; }
+
+        /// <summary>
+        /// Gets the page <see cref="TypeInfo"/>.
+        /// </summary>
+        public TypeInfo PageType { get; }
```

Comment:
Do we need this here? Isn't this redundant with the AD?

The type isn't super useful on it's own because it's always generated, I'd say just the AD should be here.